### ISSUE TITLE
getGivenName() always null?

### ISSUE THREAD

**Issue by ghost** [tamir7/Contacts#21]
The following code is resulting in contacts that never have a first name, but I know that some of them do:

```
Query q = Contacts.getQuery();
q.hasPhoneNumber();
q.include(Contact.Field.DisplayName, Contact.Field.GivenName, Contact.Field.FamilyName);
List<Contact> contacts = q.find();

for (Contact contact : contacts) {
    Log.d(TAG, "Display: " + contact.getDisplayName());
    if (contact.getGivenName() != null) {
        Log.d(TAG, "First: " + contact.getGivenName());
    }
    if (contact.getFamilyName() != null) {
        Log.d(TAG, "Last: " + contact.getFamilyName());
    }
    Log.d(TAG, "\n");
}
```

Am I missing something, or is there a bug?

**Comment by rayliverified** [comment]
Nope. getDisplayName() does not return a first name, it returns the entire First and Last name. getGivenName() does not return a first name which is most likely a bug. Instead, I get the Display Name and then subtract the Family Name to get a First Name. Here's the code I use: 

```
//Tamir7 Contacts API to fetch contacts
        Contacts.initialize(context);
        Query phonenumber = Contacts.getQuery();
        phonenumber.hasPhoneNumber();
        phonenumber.include(com.github.tamir7.contacts.Contact.Field.DisplayName,
                com.github.tamir7.contacts.Contact.Field.GivenName,
                com.github.tamir7.contacts.Contact.Field.FamilyName,
                com.github.tamir7.contacts.Contact.Field.PhoneNormalizedNumber,
                com.github.tamir7.contacts.Contact.Field.Email,
                com.github.tamir7.contacts.Contact.Field.PhotoUri);
        List<com.github.tamir7.contacts.Contact> contacts = phonenumber.find();

        for (com.github.tamir7.contacts.Contact m_contact : contacts)
        {
            Contact saveContact = new Contact();
            List<PhoneNumber> m_phonenumber = m_contact.getPhoneNumbers();
            List<Email> m_emails = m_contact.getEmails();

            //Save only first phone number and email
            if (m_phonenumber.size() >= 1)
            {
                saveContact.setPhone(ParsePhone(m_phonenumber.get(0).getNormalizedNumber()));
            }
            //Get DisplayName and edit into First Name
            if (m_contact.getDisplayName() != null)
            {
                String namecalc = m_contact.getDisplayName();
                if (m_contact.getFamilyName() != null)
                {
                    namecalc = namecalc.replace(m_contact.getFamilyName(), ""); //Remove Last Name from DisplayName
                }

                saveContact.setFirstname(namecalc);
            }
            //Overwrite DisplayName if GivenName exists.
            if (m_contact.getGivenName() != null)
            {
                saveContact.setFirstname(m_contact.getGivenName());
            }
            if (m_contact.getFamilyName() != null)
            {
                saveContact.setLastname(m_contact.getFamilyName());
            }
            if (m_emails.size() >= 1)
            {
                saveContact.setEmail(m_emails.get(0).getAddress());
            }
            if (m_contact.getPhotoUri() != null)
            {
                saveContact.setIcon(m_contact.getPhotoUri());
            }
            AddContact(saveContact);
        }

```

Hope this helps!

**Comment by rayliverified** [comment]
Fixed by PR #23. Once that PR is merged, my code above to subtract Family Name from Display Name to find Given Name is no longer necessary.

**Comment by tamir7** [comment]
Merged and released.
